### DISCUSSION TITLE
Heroku free tier will be discontinued in Nov 2023, where to move?

### DISCUSSION THREAD

**Discussion by fabriciosvallasciani** [python-telegram-bot/python-telegram-bot#3210] [Q&A] [answered]
Hi,

I´ve been hosting a telegram bot using this library for a year or so. I deployed it on Heroku free tier as I only need a database to store data and it seemed to be a quite stable place to host it. Unfortunately, Heroku free tier will be discontinued in November 2023 and I don´t know where to move my bot to. I´ve seen other suggestions here but my main concerns are:
1) what other place to host it is reliable enough to provide my bot is up and running whenever I need it. (Heroku´s limit is turning it off after 30 minutes of inactivity but that´s ok for our users.)
2) we don´t generate any income so it´s not possible to host it paying any subscription so I would need an option that is safe enough to say I won´t unexpectedly get billed EVER.

Thanks in advance.

**Comment by Bibo-Joshi** [comment, marked as answer]
Hi. The discussions section on this GitHub repo is intended for question on development and usage if the python-telegram-bot library. Discussing hosts is beyond the intended scope. Feel free to join us in the (off-topic) group on Telegram, where several people have already discussed this topic in the last days.
